File: restaurant/src/utils/grpc.py
from grpc import StatusCode

__all__ = ["grpc_status_to_http"]


grpc_http_status_mapping = {
    StatusCode.OK: 200,
    StatusCode.CANCELLED: 499,
    StatusCode.UNKNOWN: 500,
    StatusCode.INVALID_ARGUMENT: 400,
    StatusCode.DEADLINE_EXCEEDED: 504,
    StatusCode.NOT_FOUND: 404,
    StatusCode.ALREADY_EXISTS: 409,
    StatusCode.PERMISSION_DENIED: 403,
    StatusCode.UNAUTHENTICATED: 401,
    StatusCode.RESOURCE_EXHAUSTED: 429,
    StatusCode.FAILED_PRECONDITION: 400,
    StatusCode.ABORTED: 409,
    StatusCode.OUT_OF_RANGE: 400,
    StatusCode.UNIMPLEMENTED: 501,
    StatusCode.INTERNAL: 500,
    StatusCode.UNAVAILABLE: 503,
    StatusCode.DATA_LOSS: 500,
}


def grpc_status_to_http(grpc_status_code: StatusCode) -> int:
    return grpc_http_status_mapping.get(grpc_status_code, 500)  # Default to 500 if not found

File: order/src/core/utils/grpc.ts
import { Status } from "@grpc/grpc-js/build/src/constants"

Status.DATA_LOSS


const grpcHttpStatusMapping: { [key in Status]: number } = {
    [Status.OK]: 200,
    [Status.CANCELLED]: 499,
    [Status.UNKNOWN]: 500,
    [Status.INVALID_ARGUMENT]: 400,
    [Status.DEADLINE_EXCEEDED]: 504,
    [Status.NOT_FOUND]: 404,
    [Status.ALREADY_EXISTS]: 409,
    [Status.PERMISSION_DENIED]: 403,
    [Status.UNAUTHENTICATED]: 401,
    [Status.RESOURCE_EXHAUSTED]: 429,
    [Status.FAILED_PRECONDITION]: 400,
    [Status.ABORTED]: 409,
    [Status.OUT_OF_RANGE]: 400,
    [Status.UNIMPLEMENTED]: 501,
    [Status.INTERNAL]: 500,
    [Status.UNAVAILABLE]: 503,
    [Status.DATA_LOSS]: 500,
}

export function grpcStatusToHttp(grpcStatusCode: Status): number {
    // Default to 500 if not found
    return grpcHttpStatusMapping[grpcStatusCode] || 500;
}